### PR TITLE
Feature/filters

### PR DESCRIPTION
**Pull request by GoodBoyDigital**
Filter manager re-write complete
- less code
- less draw commands per filter
- any filter can have any resolution
- happy days
- bundled filters converted and tested and working

### CODE REVIEW

File: packages/core/src/renderers/filters/Filter.js [hunk 1]
```diff
@@ -92,6 +56,8 @@ export default class Filter extends Shader
          * @member {boolean}
          */
         this.autoFit = true;
+
+        this.state = new State();
```
Comment (bigtimebuddy) [inline packages/core/src/renderers/filters/Filter.js]: This could use a jsdoc

File: packages/filters/filter-fxaa/src/fxaa.frag [hunk 2]
```diff
@@ -113,11 +114,11 @@ vec4 fxaa(sampler2D tex, vec2 fragCoord, vec2 resolution,
 
 void main() {
 
-      vec2 fragCoord = vTextureCoord * filterArea.xy;
+      vec2 fragCoord = vTextureCoord * destinationFrame.ba;
```
Comment (ivanpopelyshev) [inline packages/filters/filter-fxaa/src/fxaa.frag]: Please dont use rgba fields for non-colors.

Comment (GoodBoyDigital) [reply packages/filters/filter-fxaa/src/fxaa.frag]: fair point :)

File: packages/core/src/renderers/filters/Filter.js (hunk 1 above)
Comment (GoodBoyDigital) [reply packages/core/src/renderers/filters/Filter.js]: done!

File: packages/filters/filter-fxaa/src/fxaa.frag (hunk 2 above)
Comment (GoodBoyDigital) [reply packages/filters/filter-fxaa/src/fxaa.frag]: done!

**Comment by ivanpopelyshev** [comment]
Proposal.

All global uniforms should be "filterXXXX".

1. rename `sourceFrame` to `filterFrame`
2. rename `destinationFrame` to `filterSize` or something like that, make it vec2.
3. support old `filterArea` and `filterClamp` for the sake of compatibility
4. add `filterPixel`, its value is `1.0 / filterSize / resolution`

**Comment by ivanpopelyshev** [comment]
After those changes we'll check compatibility with `pixi-filters`. @finscn, @bigtimebuddy and I made such a great job with it, and users have special filters that are based on them. 

It would be a shame if the implementation wont be compatible.

**Comment by ivanpopelyshev** [comment]
Legacy mode for filter attributes: aVertexPosition and aTextureCoord should be supported too.

**Comment by ivanpopelyshev** [comment]
filterArea shouldn't be transformed, its up to user to do that, he can override object `getBounds()` and it'll be transformed properly.

**Comment by ivanpopelyshev** [comment]
Ok, you can merge it but I fix all those problems tomorrow. We need some filters implementation, this one is fine but we cant put it in release candidate.

**Comment by GoodBoyDigital** [comment]
Auto fit added along with clamping! @ivanpopelyshev makes sense! Agree that there is a more to be done for RC. But this gets us way closer as we now have filters in v5 👍 

- The next step is to figure out the best way to get all those lovely existing filters working in v5.
- Once we have that nailed, lets PR any new features separately.

**Pull request merged**